library/layout.inc.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+* Sql functions/classes for OpenEMR.
+*
+* Things related to layout based forms in general.
+*
+* Copyright (C) 2017-2020 Rod Roark <rod@sunsetsystems.com>
+*
+* LICENSE: This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://opensource.org/licenses/gpl-license.php>.
+*
+* @package   OpenEMR
+* @link      http://www.open-emr.org
+*/
+
+// array of the data_types of the fields
+$datatypes = array(
```

Comment:
should this be identical to $datatypes in edit_layout.php (it looks different)? (if so, then prob better to centralize it so don't have to update both of them)

Reply:
It will be centralized in a commit coming up.